core/src/net/mod.rs

```diff
@@ -54,6 +55,34 @@ impl From<Transport> for Protocol {
     }
 }
 
+/// Session identifier, part of the `NetMessage` struct. Managed by Kompact internally, may be read
+/// by users to detect session loss, indicated by different SessionId's of NetMessages.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct SessionId {
+    session: Uuid,
+}
+
+impl SessionId {
+    /// Generates a new SessionId
+    pub(crate) fn new() -> SessionId {
+        SessionId {
+            session: Uuid::new_v4(),
+        }
+    }
```

Comment:
Again this makes it impossible to use in 3rd party impls.

Reply:
Also shall we call this `new_random` or `new_unique` to make it clear at use-site that calling this twice will give different results?